packages/webapp/src/containers/Animals/Inventory/useAnimalInventory.tsx

```diff
@@ -0,0 +1,282 @@
+/*
+ *  Copyright 2024 LiteFarm.org
+ *  This file is part of LiteFarm.
+ *
+ *  LiteFarm is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  LiteFarm is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ *  GNU General Public License for more details, see <https://www.gnu.org/licenses/>.
+ */
+import { FC, useMemo } from 'react';
+import i18n from '../../../locales/i18n';
+import {
+  useGetAnimalsQuery,
+  useGetAnimalBatchesQuery,
+  useGetAnimalGroupsQuery,
+  useGetCustomAnimalBreedsQuery,
+  useGetCustomAnimalTypesQuery,
+  useGetDefaultAnimalBreedsQuery,
+  useGetDefaultAnimalTypesQuery,
+} from '../../../store/api/apiSlice';
+import useQueries from '../../../hooks/api/useQueries';
+import {
+  Animal,
+  AnimalBatch,
+  AnimalGroup,
+  CustomAnimalBreed,
+  CustomAnimalType,
+  DefaultAnimalBreed,
+  DefaultAnimalType,
+} from '../../../store/api/types';
+import { getComparator, orderEnum } from '../../../util/sort';
+import { ReactComponent as CattleIcon } from '../../../assets/images/animals/cattle-icon.svg';
+import { ReactComponent as ChickenIcon } from '../../../assets/images/animals/chicken-icon.svg';
+import { ReactComponent as PigIcon } from '../../../assets/images/animals/pig-icon.svg';
+import { ReactComponent as BatchIcon } from '../../../assets/images/animals/batch.svg';
+import { AnimalTranslationKey } from '../types';
+
+export type AnimalInventory = {
+  icon: FC;
+  identification: string;
+  type: string;
+  breed: string;
+  groups: string[];
+  path: string;
+  count: number;
+  batch: boolean;
+};
+
+const { t } = i18n;
+
+const getDefaultAnimalIcon = (
+  defaultAnimalTypes: DefaultAnimalType[],
+  defaultTypeId: number | null,
+) => {
+  const key = defaultAnimalTypes.find(({ id }) => id === defaultTypeId)?.key;
+  switch (key) {
+    case AnimalTranslationKey.CATTLE:
+      return CattleIcon;
+    case AnimalTranslationKey.CHICKEN_BROILERS:
+      return ChickenIcon;
+    case AnimalTranslationKey.CHICKEN_LAYERS:
+      return ChickenIcon;
+    case AnimalTranslationKey.PIGS:
+      return PigIcon;
+    default:
+      return CattleIcon;
```

Comment:
Sorry I forgot to tell you, but for custom types, Loïc mentioned that we should use the animal icon used for the menu. I tried using the new icon that can be found in the mockup, but it wasn't visible unfortunately. https://www.figma.com/file/eMffC0h3JPlWtzNttxJpil?node-id=3089:43641&mode=design#718160629

Reply:
Yeah I think I still need to receive an animal Icon that matches these icons color for the menu too. 

I might end up manually editing the stroke or fill color of the svg to do so.